extensions/common/iam/identity-trust/identity-trust-transform/src/main/java/org/eclipse/edc/iam/identitytrust/transform/to/JsonObjectToIssuerTransformer.java

```diff
@@ -0,0 +1,41 @@
+/*
+ *  Copyright (c) 2023 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - initial API and implementation
+ *
+ */
+
+package org.eclipse.edc.iam.identitytrust.transform.to;
+
+import jakarta.json.JsonObject;
+import org.eclipse.edc.identitytrust.model.Issuer;
+import org.eclipse.edc.jsonld.spi.transformer.AbstractJsonLdTransformer;
+import org.eclipse.edc.transform.spi.TransformerContext;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.HashMap;
+
+public class JsonObjectToIssuerTransformer extends AbstractJsonLdTransformer<JsonObject, Issuer> {
+    protected JsonObjectToIssuerTransformer() {
+        super(JsonObject.class, Issuer.class);
+    }
+
+    @Override
+    public @Nullable Issuer transform(@NotNull JsonObject jsonObject, @NotNull TransformerContext context) {
+        var id = nodeId(jsonObject);
+        var props = new HashMap<String, Object>();
+        visitProperties(jsonObject, (key, jsonValue) -> {
+            var res = transformGenericProperty(jsonValue, context);
+            props.put(key, res);
+        });
+        return new Issuer(id, props);
+    }
```

Comment:
Might deserve few unit-tests. 

Also do we want the id to also be put in the `additionalProperties` map? I would say no but seems this is what current implem is doing

Reply:
Yes, and I just remembered that the transformer also needs a few tests...

Reply:
Assuming we have validation on the ingress, we can assume that the issuer (mandatory field) is there, and since the jsonld is expanded, the array can't be empty